typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -324,20 +325,25 @@
   (define-syntax-class name-exists-kw
     (pattern #:name-exists))
   (syntax-parse stx
+    [_ #:when (eq? 'module-begin (syntax-local-context))
+       ;; it would be inconvenient to find the correct #%module-begin here, so we rely on splicing
+       #`(begin #,stx (begin))]
     [(_ ty:id pred:id lib (~optional ne:name-exists-kw) ...)
      (with-syntax ([hidden (generate-temporary #'pred)])
+       (define pred-cnt
+         (syntax-local-lift-expression
+          (make-contract-def-rhs #'(-> Any Boolean) #f #f)))
```

Comment:
No, because we say that the type of the predicate is `(Any -> Boolean)`.  The generated contract and the type need to be consistent.

Changing the type to be `(-> Any Any)` would break programs like this:

```
(require/typed m [#:opaque T t?])
(define x : Boolean (t? #t))
```

Also, keeping the more restrictive type can't break existing programs because we previously used `boolean?` in the contract.